cdap-ui/app/cdap/components/Replicator/utilities/index.ts

```diff
@@ -246,6 +232,142 @@ export function constructTablesSelection(tables, columns, dmlBlacklist) {
   return tablesArr;
 }
 
+export async function convertConfigToState(rawConfig, parentArtifact) {
+  const newState: Partial<ICreateState> = {
+    loading: false,
+    name: rawConfig.label || rawConfig.name || '',
+    description: objectQuery(rawConfig, 'config', 'description') || '',
+    activeStep: 1,
+    offsetBasePath: objectQuery(rawConfig, 'config', 'offsetBasePath') || '',
+    numInstances: objectQuery(rawConfig, 'config', 'parallelism', 'numInstances') || 1,
+  };
+
+  const stages = objectQuery(rawConfig, 'config', 'stages') || [];
+
+  // SOURCE
+  const source = stages.find((stage) => {
+    const stageType = objectQuery(stage, 'plugin', 'type');
+    return stageType === PluginType.source;
+  });
+
+  if (source) {
+    const sourceArtifact = objectQuery(source, 'plugin', 'artifact') || {};
+
+    const sourcePluginInfo = await fetchPluginInfo(
+      parentArtifact,
+      sourceArtifact.name,
+      sourceArtifact.scope,
+      source.plugin.name,
+      source.plugin.type
+    ).toPromise();
+
+    newState.sourcePluginInfo = sourcePluginInfo;
+    newState.sourceConfig = objectQuery(source, 'plugin', 'properties') || {};
+
+    try {
+      const sourcePluginWidget = await fetchPluginWidget(
+        sourceArtifact.name,
+        sourceArtifact.version,
+        sourceArtifact.scope,
+        sourcePluginInfo.name,
+        sourcePluginInfo.type
+      ).toPromise();
+
+      newState.sourcePluginWidget = sourcePluginWidget;
+    } catch (e) {
+      // tslint:disable-next-line: no-console
+      console.log('Cannot fetch source plugin widget', e);
```

Comment:
At this point should we surface the error to the user? This will happen when the config is not valid or if the plugin has been deleted before the clone?

Reply:
I don't think that's necessary, this error happens when the widget json is not present (which is not required)

Reply:
Would this mean we show just plain text boxes instead of widgets? Shouldn't that be surfaced to the user similar to the pipeline?

Reply:
Talked to @elfenheart. We stopped showing the warning even in pipelines. This is fine to maintain consistency.